src/test/scala/com/gradle/internal/UtilsTest.scala

```diff
@@ -1,3 +1,46 @@
 package com.gradle.internal
 
-class UtilsTest {}
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.prop.TableDrivenPropertyChecks
+
+import java.net.URI
+
+class UtilsTest extends AnyFlatSpec
+  with Matchers
+  with TableDrivenPropertyChecks {
+
+  "Utils" should "construct a repo URI from a git URL" in {
+    forEvery(webRepoUriArgumentsTable) { (repositoryUri, expectedWebRepoUri) =>
+      assertResult(Some(URI.create(expectedWebRepoUri))) (Utils.toWebRepoUri(repositoryUri))
+    }
+  }
+
+  private lazy val webRepoUriArgumentsTable = Table((
+    "remote repository", "expected resolved repository"),
+    (for (host <- hosts; repo <- repos) yield (replaceHost(repo._1, host), replaceHost(repo._2, host))): _*
+  )
```

Comment:
```suggestion
  private lazy val webRepoUriArgumentsTable = {
    val rows = for (host <- hosts; repo <- repos) yield (replaceHost(repo._1, host), replaceHost(repo._2, host))
    Table(
      ("remote repository", "expected resolved repository"),
      rows: _*
    )
  }
```

I think this'd be easier to read with an intermediate variable.